### DISCUSSION TITLE
Stretch facets

### DISCUSSION THREAD

**Discussion by adrimg89** [cwapi3d/cwapi3dpython#220] [Q&A]
Hello everybody.

Does anyone know if is possible to stretch a facet which is not start or end facet with the api?

Thank you in advance

**Comment by Antoine-S-C** [comment]
Hi,
An idea, if you have to stretch a width or height facet :
- rotate the element length axis with geometry_controller.rotate_length_axis_90(elements: List[int])
- stretch the new start or end facet
- put the length axis back as it was at the beginning

**Reply by adrimg89** [reply]
Hi!!

Thank you very much for your reply.

The option that you propose I am already doing when is needed, but my case is very different: what can I do when one of the sides of the panel has a central cut which made two paralel front faces? 
Teorically both faces are start or end face, but the program only understand as start face one of them.

**Reply by Antoine-S-C** [reply]
I'm not sure I understand the shape of your panel.
Have you tried to do with an element that is the complementary shape to your panel (= the cut). This element has its own start and end facets that you can stretch. Either to remove volume from your original panel or to add some to it.

**Reply by Brunner246** [reply]
@adrimg89 
maybe you could try something similar to this https://github.com/cwapi3d/cwapi3dpython/discussions/147#discussioncomment-10771486

**Reply by adrimg89** [reply]
Thank you @Brunner246 . I'll study the answer.
